File: 867.transpose-matrix.0.kt
/*
 * @lc app=leetcode id=867 lang=kotlin
 *
 * [867] Transpose Matrix
 */
class Solution {
    fun transpose(A: Array<IntArray>): Array<IntArray> {
        val cols = A[0].size

        return A.fold(List<IntArray>(cols) { IntArray(0) } ) { list, arr ->
            list.zip(arr.toList()) { a, b -> a + b }
        }.toTypedArray()
    }
}

File: 867.transpose-matrix.0.go
/*
 * @lc app=leetcode id=867 lang=golang
 *
 * [867] Transpose Matrix
 */
func transpose(A [][]int) [][]int {
    r := len(A)
    c := len(A[0])
    ret := make([][]int, c)
    for i := 0; i < c; i++ {
        ret[i] = make([]int, r)
    }

    for i := 0; i < r; i++ {
        for j := 0; j < c; j++ {
            ret[j][i] = A[i][j]
        }
    }
    return ret
}
